OpenSim/Simulation/Test/testInverseKinematicsSolver.cpp

```diff
@@ -0,0 +1,457 @@
+ * OpenSim is developed at Stanford University and supported by the US        *
+ * National Institutes of Health (U54 GM072970, R24 HD065690) and by DARPA    *
+ * through the Warrior Web program.                                           *
+ *                                                                            *
+ * Copyright (c) 2005-2016 Stanford University and the Authors                *
+ * Author(s): Ajay Seth                                                       *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+//=============================================================================
+// testInverseKinematicsSolver verifies that changes to the accuracy and marker
+// weights have expected effects on the inverse kinematics results/errors
+//=============================================================================
+#include <OpenSim/Simulation/osimSimulation.h>
+#include <OpenSim/Simulation/InverseKinematicsSolver.h>
+#include <OpenSim/Simulation/MarkersReference.h>
+#include <OpenSim/Common/Constant.h>
+#include <OpenSim/Common/STOFileAdapter.h>
+#include <random>
+
+using namespace OpenSim;
+using namespace std;
+
+// Utility function to build a simple pendulum with markers attached
+Model* constructPendulumWithMarkers();
+// Using a model with markers and trajectory of states, create synthetic
+// marker data. If noiseRadius is provided use it to scale the noise
+// that perturbs the marker data. Optional fixed parameter to use the same
+// noise for each time frame or to compute noise to be added at each frame.
+MarkerData* generateMarkerDataFromModelAndStates(const Model& model,
+    const StatesTrajectory& states, double noiseRadius=0, bool fixed=false);
+
+// Verify that accuracy improves the number of decimals points to which
+// the solver solution (coordinates) can be trusted as it is tightened.
+void testAccuracy();
+// Verify that the marker weights impact the solver and has the expected
+// effect of reducing the error for the marker weight that is increased. 
+void testUpdateMarkerWeights();
+// Verify that the track() solution is also effected by updating marker
```

Comment:
Yes, it is correct to say "we verify the *effect* on the solution", but on L53 the word "effected" is used as a verb and therefore means "caused to happen" AFAIK. The comment is saying that updating marker weights *caused* the track() solution, which I don't think makes sense because a solution is computed, not *caused* (*effected*).